src/pages/index.js

```diff
@@ -3,104 +3,248 @@ import FormValidator from "../components/FormValidator.js";
+
+  api
+    .toggleLike(cardId, isLiked)
+    .then((updatedCard) => {
+      if (updatedCard && "isLiked" in updatedCard) {
+        likeButton.classList.toggle(
+          "card__like-button_active",
+          updatedCard.isLiked
+        );
+      }
+    })
+    .catch((error) => console.error("Error updating like:", error));
 }
 
 /*Event Handler*/
 
-function handleProfileEditSubmit(inputValue) {
-    userInfo.setUserInfo({
-        name: inputValue.title,
-        description: inputValue.description,
+function handleProfileEditSubmit(event) {
+  const inputValues = popupWithEditProfileForm.getInputValues(); // Get form values
+
+  if (!inputValues.name || !inputValues.about) {
+    return;
+  }
+
+  popupWithEditProfileForm.setLoadingText(true);
+
+  api
+    .updateUserInfo(inputValues.name, inputValues.about)
+    .then((updatedUser) => {
+      userInfo.setUserInfo(updatedUser);
+      popupWithEditProfileForm.close();
+    })
+    .catch((err) => console.error("Profile update failed:", err))
+    .finally(() => {
+      popupWithEditProfileForm.setLoadingText(false);
+    });
+}
+
+function handleAvatarUpdateSubmit(event) {
+  const inputvalues = popupWithEditAvatarForm.getInputValues(); // Get form values
+  const avatarUrl = inputvalues.avatar;
+
+  if (!avatarUrl) {
+    return;
+  }
+
+  popupWithEditAvatarForm.setLoadingText(true);
+  api
+    .updateProfileAvatar(avatarUrl)
+    .then((res) => {
+      userInfo.setUserAvatar(res.avatar);
+      popupWithEditAvatarForm.close();
+    })
+    .catch((err) => console.error("Profile update failed:", err))
+    .finally(() => {
+      popupWithEditAvatarForm.setLoadingText(false);
     });
-    popupWithEditProfileForm.close();
-    editFormValidator.disableSubmitButton();
 }
 
 function handleAddCardFormSubmit(inputValue) {
-    const cardData = {
-        name: inputValue.title,
-        link: inputValue.url,
-    };
-    renderCard(cardData);
-    popupWithAddCardForm.close();
-    addCardForm.reset();
-    addFormValidator.disableSubmitButton();
+  const cardData = {
+    name: inputValue.title,
+    link: inputValue.url,
+  };
+
+  popupWithAddCardForm.setLoadingText(true);
+  api
+    .addNewCard(cardData)
+    .then((newCard) => {
+      renderCard(newCard);
+    })
+    .finally(() => {
+      popupWithAddCardForm.setLoadingText(false);
+    });
+  popupWithAddCardForm.close();
+  addCardForm.reset();
+  addFormValidator.disableSubmitButton();
+}
 
+function openConfirmPopup(card) {
+  popupWithDeleteButton.setSubmitFunction(() => {
+    // this should be basically the handleDeleteCardConfirm logic
+    api
+      .deleteCardData(card._cardId)
+      .then(() => {
+        // use handleDeleteCard which removes the card from the DOM
+        card.handleDeleteCard(); //
+      })
+      .catch((err) => {
+        console.error("Failed to delete card: ", err);
+      })
+      .finally(() => {
+        popupWithDeleteButton.close();
```

Comment:
* You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well. **Please, correct it everywhere in the code**